nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/AddPermissionTest.java

```diff
@@ -421,6 +426,37 @@ public void shouldAddPermissionForUserWhenBlockingInheritance() throws Operation
         assertExpectedPermissions(acl, params, "existingUser1");
     }
 
+    @Test
+    public void shouldAddPermissionForUsersWhenUsingChainExtension() throws OperationException {
+        // Mock the users params
+        List<String> users = Arrays.asList("jchastain", "rdeniro", "dwashington");
+        users.forEach(user -> when(userManager.getUserModel(user)).thenReturn(new SimpleDocumentModel(user)));
+
+        DocumentModel doc = session.getDocument(new PathRef("/src"));
+        OperationContext ctx = new OperationContext(session);
+        ctx.setInput(doc);
+
+        Object object = automationService.run(ctx, "testAddPermissionToUsers");
+        assertThat(object, instanceOf(DocumentModel.class));
+
+        DocumentModel documentModel = (DocumentModel) object;
+
+        ACL acl = documentModel.getACP().getACL("local");
+        assertNotNull("Local ACL cannot be null !", acl);
+        assertThat("ACEs size must be the same then users size !", acl.getACEs(), arrayWithSize(users.size()));
+
+        Map<String, ACE> mapAceByUser = Arrays.stream(acl.getACEs())
+                                              .collect(Collectors.toMap(ACE::getUsername, ace -> ace));
```

Comment:
Maybe use `Function.identity()`?

Reply:
Hello,
Function.identity() in this case returns the ACE, using Function.identity() will gives me a Map with ACE as key. But i want a user name as the key because my input is an array of users.

Reply:
I was just thinking of replacing `ace -> ace`:
```Collectors.toMap(ACE::getUsername, Function.identity())```

Reply:
This is a test, we don't care about performance, use whatever is shorter / most understandable. `ace -> ace` is fine. And anyway in the end `Function.identity()` and `ace -> ace` will very likely be optimized in the same way by the JDK JIT compiler. JDK developers themselves recommend to use just `foo -> foo` instead of `Function.identity()`.

Reply:
yep, i agree